libs/user-facing-errors/src/schema_engine.rs

```diff
@@ -267,6 +267,20 @@ pub struct ForeignKeyCreationNotAllowed;
 )]
 pub struct DirectDdlNotAllowed;
 
+#[derive(Debug, SimpleUserFacingError)]
```

Comment:
These were brought over from the deleted `introspection_engine.rs` right? We're missing code `4000`?
> Introspection operation failed to produce a schema file

Reply:
The code 4000 is not used anywhere in the codebase.

Reply:
Cool so this was intentionally removed as clean-up, good to know :) Need to double check if this still exists in prisma/prisma too